### PR TITLE
build: Add `show-%` target for multi-line variables and debug info

### PR DESCRIPTION
**Pull request by hebasto**
This PR is a continuation of bitcoin/bitcoin#17087.

Our depends builds system uses a bunch of variables which values consist of some lines, i.e., multi-line variables.

During debugging it is annoying that `print-%` is unable to handle such variables:
```
$ cd depends
$ make print-sqlite_set_vars
/bin/sh: 1: Syntax error: Unterminated quoted string
make: *** [Makefile:5: print-sqlite_set_vars] Error 2
```

This PR adds `show-%` target which prints multi-line variable value and its origin and flavor as well:
```
$ cd depends
$ make show-sqlite_set_vars
- name:   sqlite_set_vars
- origin: file
- flavor: simple
- value, printed from the new line:
sqlite_config_opts=--disable-shared --disable-readline --disable-dynamic-extensions --enable-option-checking
sqlite_config_opts_linux=--with-pic
sqlite_config_opts_freebsd=--with-pic
sqlite_config_opts_netbsd=--with-pic
sqlite_config_opts_openbsd=--with-pic
```

For connoisseurs suggesting `make show-qt_set_vars` :)

---

Other useful examples of usage:
```
$ make show-CC
- name:   CC
- origin: default
- flavor: recursive
- value, printed from the new line:
cc
```

```
$ make show-CC CC=clang
- name:   CC
- origin: command line
- flavor: recursive
- value, printed from the new line:
clang
```

```
$ CC=clang make show-CC
- name:   CC
- origin: environment
- flavor: recursive
- value, printed from the new line:
clang
```

```
$ make show-host
- name:   host
- origin: file
- flavor: simple
- value, printed from the new line:
x86_64-pc-linux-gnu
```

---

Please note that it is unwanted to just modify the `print-%` rule as it is used in [scripts](https://github.com/bitcoin/bitcoin/blob/5034b7fa3b97c1cbaf93b337d462777f3e25d256/contrib/guix/guix-build#L130).

### CODE REVIEW

**Comment by laanwj** [comment]
Any reason for keeping this in draft status?

**Comment by jonatack** [comment]
Concept ACK

**Comment by hebasto** [comment]
@laanwj 
> Any reason for keeping this in draft status?

The wrong output here:
```
$ make show-sqlite_set_vars
- name:   sqlite_set_vars
- origin: file
- flavor: recursive
- value, printed from the new line:
zeromq_config_opts=--disable-shared --disable-readline --disable-dynamic-extensions --enable-option-checking
zeromq_config_opts_linux=--with-pic
zeromq_config_opts_freebsd=--with-pic
zeromq_config_opts_netbsd=--with-pic
zeromq_config_opts_openbsd=--with-pic
```

**Comment by hebasto** [comment]
> Any reason for keeping this in draft status?

Ready for reviewing now.

@laanwj 

Thank you for your reminding.

**Comment by hebasto** [comment]
The PR description has been updated.

**Comment by jarolrod** [comment]
concept ack

File: depends/packages/bdb.mk [hunk 1]
```diff
@@ -6,7 +6,7 @@ $(package)_sha256_hash=12edc0df75bf9abd7f82f821795bcee50f42cb2e5f76a6a281b857327
 $(package)_build_subdir=build_unix
 $(package)_patches=clang_cxx_11.patch
 
-define $(package)_set_vars
+define $(package)_set_vars :=
```
Comment (laanwj) [inline depends/packages/bdb.mk:9]: Maybe a dumb question, but what does the added `:=` do?

**Comment by hebasto** [comment]
Rebased f323829527dfefd3d108c23983a7c60f8edd20e8 -> 2d290d8a4a2fe3d620bfbaf5c3b719401288a87b ([pr24283.03](https://github.com/hebasto/bitcoin/commits/pr24283.03) -> [pr24283.04](https://github.com/hebasto/bitcoin/commits/pr24283.04)).

**Comment by hebasto** [comment]
Rebased 2d290d8a4a2fe3d620bfbaf5c3b719401288a87b -> 6c18d85889ebdd70241d69023d2dade8a2270efc ([pr24283.04](https://github.com/hebasto/bitcoin/commits/pr24283.04) -> [pr24283.05](https://github.com/hebasto/bitcoin/commits/pr24283.05)) due to the conflict with #25697.

**Comment by hebasto** [comment]
Rebased 6c18d85889ebdd70241d69023d2dade8a2270efc -> 9d568cceca18c00100a3ed9bd735477553501426 ([pr24283.05](https://github.com/hebasto/bitcoin/commits/pr24283.05) -> [pr24283.06](https://github.com/hebasto/bitcoin/commits/pr24283.06)) due to the conflict with #25633.

**Comment by achow101** [comment]
Are you still working on this?

**Comment by hebasto** [comment]
> Are you still working on this?

I am. However, closing due to the lack of interest from other developers.

**Pull request closed without merging**